src/test/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionSpec.scala

```diff
@@ -0,0 +1,143 @@
+import org.scalatest.{BeforeAndAfterAll, Matchers, FlatSpecLike}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+/**
+ * Created by dvoet on 5/17/16.
+ */
+class WorkflowSubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpecLike with Matchers with TestDriverComponent with BeforeAndAfterAll {
+  import driver.api._
+
+  def this() = this(ActorSystem("WorkflowSubmissionSpec"))
+  val mockGoogleServicesDAO: MockGoogleServicesDAO = new MockGoogleServicesDAO("test")
+
+  class TestWorkflowSubmission(
+    val dataSource: SlickDataSource,
+    val batchSize: Int = 3, // the mock remote server always returns 3, 2 success and an error
+    val pollInterval: FiniteDuration = 1 second) extends WorkflowSubmission {
+
+    val mockServer = RemoteServicesMockServer()
+
+    val credential: Credential = new MockGoogleCredential.Builder().build()
+    val googleServicesDAO = mockGoogleServicesDAO
+    val executionServiceDAO: ExecutionServiceDAO = new HttpExecutionServiceDAO(mockServer.mockServerBaseUrl, mockServer.defaultWorkflowSubmissionTimeout)
+    val methodRepoDAO = new HttpMethodRepoDAO(mockServer.mockServerBaseUrl)
+  }
+
+  class TestWorkflowSubmissionWithMockExecSvc(
+     dataSource: SlickDataSource,
+     batchSize: Int = 3, // the mock remote server always returns 3, 2 success and an error
+     pollInterval: FiniteDuration = 1 second) extends TestWorkflowSubmission(dataSource, batchSize, pollInterval) {
+    override val executionServiceDAO = new MockExecutionServiceDAO()
+  }
+
+  override def beforeAll = {
+    super.beforeAll
+    Await.result( mockGoogleServicesDAO.storeToken(userInfo, UUID.randomUUID.toString), Duration.Inf )
+  }
+
+  "WorkflowSubmission" should "get a batch of workflows" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource)
+
+    val workflowRecs = runAndWait(
+      for {
+        workflowRecs <- workflowQuery.findWorkflowsBySubmissionId(UUID.fromString(testData.submission1.submissionId)).take(workflowSubmission.batchSize).result
+        _ <- workflowQuery.batchUpdateStatus(workflowRecs, WorkflowStatuses.Queued)
+      } yield {
+        workflowRecs
+      }
+    )
+
+    assertResult(SubmitWorkflowBatch(workflowRecs.map(_.id))) {
+      Await.result(workflowSubmission.getUnlaunchedWorkflowBatch(), Duration.Inf)
+    }
+
+    assert(runAndWait(workflowQuery.findWorkflowByIds(workflowRecs.map(_.id)).result).forall(_.status == WorkflowStatuses.Launching.toString))
+  }
+
+  it should "schedule the next check when there are no workflows" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource)
+    assertResult(ScheduleNextWorkflowQuery) {
+      Await.result(workflowSubmission.getUnlaunchedWorkflowBatch(), Duration.Inf)
+    }
+  }
+
+  it should "have only 1 submission in a batch" in withDefaultTestDatabase {
```

Comment:
batches of workflows submitted to cromwell must all belong to the same submission (because they all have to share the same wdl). this tests that